src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -103,19 +106,31 @@ class CompilerContainer {
    */
   compilationButton (name) {
     if (!name) name = ''
-    var displayed = name === '' ? '<no file selected>' : name
-    var el = yo`
-    <div class="px-2 mt-2 pb-0 d-flex">
-      <button class="btn btn-primary btn-block ${name === '' ? 'disabled' : ''}" title="Compile" onclick="${this.compile.bind(this)}">
+    const displayed = name === '' ? '<no file selected>' : name
+    const disabled = name === '' ? 'disabled' : ''
+    const compileBtn = yo`
+      <button id="compileBtn" class="btn btn-primary btn-block ${disabled}" title="Compile" onclick="${this.compile.bind(this)}">
         <span>${this._view.compileIcon} Compile ${displayed}</span>
       </button>
-    </div>`
-    if (name === '') {
-      el.setAttribute('disabled', 'true')
-    }
+    `
+    let el = yo`
+      <div class="px-2 mt-2 pb-0 d-flex">
+        ${compileBtn}
+      </div>
+    `
     return el
   }
 
+  _disableCompileBtn (disable) {
+    let btn = document.querySelector("button[id='compileBtn']")
```

Comment:
Id are supposed to be unique, better to do: `document.getElementById('compileBtn')` (it's much faster)